### PR TITLE
[Web] Audio

### PR DESCRIPTION
**Pull request by EvanBacon**
# How

* Added a global emitter (expo-sensors) for events and callbacks
* Extended what was already in place for video

### CODE REVIEW

File: packages/expo-av/src/Audio/Sound.ts [hunk 1]
```diff
@@ -12,10 +12,11 @@ import {
 } from '../AV';
 import ExponentAV from '../ExponentAV';
 
+type AudioInstance = number | HTMLMediaElement;
 export class Sound implements Playback {
   _loaded: boolean = false;
   _loading: boolean = false;
-  _key: number = -1;
+  _key?: AudioInstance;
```
Comment (ide) [inline packages/expo-av/src/Audio/Sound.ts]: Like the other properties below:

```suggestion
  _key: AudioInstance | null = null;
```

File: packages/expo-av/src/GlobalPlatformEmitter.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import RCTDeviceEventEmitter from 'react-native/Libraries/EventEmitter/RCTDeviceEventEmitter';
```
Comment (ide) [inline packages/expo-av/src/GlobalPlatformEmitter.ts]: Can you open up a PR to move this module to expo-react-native-adapter and export it through expo-core?

**Review by ide — CHANGES_REQUESTED** [review]

File: packages/expo-av/src/GlobalPlatformEmitter.ts (hunk 2 above)
Comment (EvanBacon) [reply packages/expo-av/src/GlobalPlatformEmitter.ts]: #3368

File: packages/expo-av/src/ExponentAV.web.ts [hunk 3]
```diff
@@ -39,80 +40,118 @@ function getStatusFromMedia(media?: HTMLMediaElement): PlaybackStatus {
...
+      });
+    };
 
-  async getStatusForSound() {},
-  async setErrorCallbackForSound() {},
-  async loadForSound() {},
-  async unloadForSound() {},
-  async setStatusForSound() {},
-  async replaySound() {},
+    media.onerror = (event: Event | string, source?: string, fileno?: number, columnNumber?: number, error?: Error) => {
```
Comment (ide) [inline packages/expo-av/src/ExponentAV.web.ts]: The properties event, source, fileno, etc... are properties of ErrorEvent objects -> not sure these get passed in as separate arguments. Also for media elements (video, audio), I believe `media.error` gets set when there's an `error` event, and we could reliably get a JS error object from there:

```js
media.onerror = () => {
  // media.error.message and media.error.code are defined
  // https://developer.mozilla.org/en-US/docs/Web/API/MediaError
};
```

**Review by ide — APPROVED** [review]

**Pull request merged**